### PR TITLE
Prereqs for validating service instance

### PR DESCRIPTION
**Pull request by wryun**
Scaffolding (informers) set up, no actual validation yet. Smaller PRs?

Part of #187

### CODE REVIEW

File: cmd/smith/app/app.go [hunk 1]
```diff
@@ -117,6 +118,10 @@ func (a *App) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	var catalogger *catalog.Catalog
```
Comment (wryun) [inline cmd/smith/app/app.go]: I'm really unhappy with the naming for this one, but I'm not sure what it should be.

Comment (amckague) [reply cmd/smith/app/app.go]: hmm, var catalogClient *catalog.Client is go ish?

File: pkg/controller_test/zz_plumbing_for_test.go [hunk 2]
```diff
@@ -148,13 +163,49 @@ func (tc *testCase) run(t *testing.T) {
...
+					},
+					Spec: sc_v1b1.ClusterServicePlanSpec{
+						ClusterServiceClassRef: sc_v1b1.ClusterObjectReference{
+							Name: serviceClassName,
+						},
+						ExternalName: servicePlanExternalName,
+						ExternalID:   servicePlanName,
+					},
+				},
+			}...)...,
```
Comment (amckague) [inline pkg/controller_test/zz_plumbing_for_test.go]: more dots

Comment (wryun) [reply pkg/controller_test/zz_plumbing_for_test.go]: yeah... this was a bit silly. I was a little bit too keen to write it as one expression, I guess.

**Comment by wryun** [comment]
I think I've addressed what we talked about @ash2k , but I probably missed something...

**Pull request merged**